tests/jobs/test_scheduler_job.py

```diff
@@ -3920,6 +3920,50 @@ def test_do_schedule_max_active_runs_task_removed(self):
         ti = run1.get_task_instance(task1.task_id, session)
         assert ti.state == State.QUEUED
 
+    def test_do_schedule_max_active_runs_task_shutdown(self):
+        """Test that tasks in shutdown state don't count as actively running."""
```

Comment:
I'm not so sure about this. While they are in Shutdown state _they are still running_.